internal/service/advancedcluster/plural_data_source.go

```diff
@@ -84,6 +84,7 @@ func (d *pluralDS) getBasicClusters(ctx context.Context, diags *diag.Diagnostics
 	})
 	if err != nil {
 		addListError(diags, projectID, err)
+		RemoveClusterNotFoundErrors(diags)
```

Comment:
related to Espen comment: Given this is a fix in behaviour of the plural data source would adjust PR title and add a changelog entry